tests/bsim/bluetooth/audio/src/bap_scan_delegator_test.c

```diff
@@ -5,76 +5,830 @@
+		return true;
+	}
+
+	if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO) != 0) {
+		return true;
+	}
+
+	g_broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+	bt_addr_le_to_str(info->addr, addr_str, sizeof(addr_str));
+
+	printk("Found BAP broadcast source with address %s and ID 0x%06X\n",
+	       addr_str, g_broadcast_id);
+
+	state = sync_state_get_or_new(NULL);
+	if (state == NULL) {
+		FAIL("Failed to get sync state");
+		return true;
+	}
+
+	printk("Creating Periodic Advertising Sync\n");
+	bt_addr_le_copy(&sync_create_param.addr, info->addr);
+	sync_create_param.sid = info->sid;
+	sync_create_param.timeout = 0xa;
+
+	err = bt_le_per_adv_sync_create(&sync_create_param,
+					&state->pa_sync);
+	if (err != 0) {
+		FAIL("Could not create PA sync (err %d)\n", err);
+		return true;
+	}
+
+	state->pa_syncing = true;
+
+	return false;
 }
 
-static struct bt_bap_scan_delegator_cb scan_delegator_cb = {
-	.pa_synced = pa_synced,
-	.pa_term = pa_term,
-	.pa_recv = pa_recv
+static void scan_recv_cb(const struct bt_le_scan_recv_info *info,
+			 struct net_buf_simple *ad)
+{
+	/* We are only interested in non-connectable periodic advertisers */
+	if ((info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) != 0 ||
+	    info->interval == 0) {
+		return;
+	}
+
+	bt_data_parse(ad, broadcast_source_found, (void *)info);
+}
+
+static struct bt_le_scan_cb scan_cb = {
+	.recv = scan_recv_cb
 };
 
-static void test_main(void)
+static int add_source(struct sync_state *state)
+{
+	struct bt_bap_scan_delegator_add_src_param param;
+	int res;
+
+	UNSET_FLAG(flag_recv_state_updated);
+
+	param.pa_sync = state->pa_sync;
+	param.encrypt_state = BT_BAP_BIG_ENC_STATE_NO_ENC;
+	param.broadcast_id = g_broadcast_id;
+	param.num_subgroups = 1U;
+
+	for (uint8_t i = 0U; i < param.num_subgroups; i++) {
+		struct bt_bap_scan_delegator_subgroup *subgroup_param = &param.subgroups[i];
+
+		subgroup_param->bis_sync = BT_BAP_BIS_SYNC_NO_PREF;
+		subgroup_param->metadata_len = 0U;
+		(void)memset(subgroup_param->metadata, 0, sizeof(subgroup_param->metadata));
+	}
+
+	res = bt_bap_scan_delegator_add_src(&param);
+	if (res < 0) {
+		return res;
+	}
+
+	state->src_id = (uint8_t)res;
+
+	WAIT_FOR_FLAG(flag_recv_state_updated);
+
+	return res;
+}
+
+static void add_all_sources(void)
+{
+	for (size_t i = 0U; i < ARRAY_SIZE(sync_states); i++) {
+		struct sync_state *state = &sync_states[i];
+
+		if (state->pa_sync != NULL) {
+			int res;
+
+			printk("[%zu]: Adding source\n", i);
+
+			res = add_source(state);
+			if (res < 0) {
+				FAIL("[%zu]: Add source failed (err %d)\n", i, res);
+				return;
+			}
+
+			printk("[%zu]: Source added with id %u\n",
+			       i, state->src_id);
+		}
+	}
+}
+
+static int mod_source(struct sync_state *state)
+{
+	const uint16_t pref_context = BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL;
+	struct bt_bap_scan_delegator_mod_src_param param;
+	uint8_t pref_context_metadata[4] = {
+		0x03,
```

Comment:
I know those are just tests, but maybe it will be good to add a comment describing this 0x03 value.
I.e is this just a random number within metadata, or does this represent sth?

Reply:
It is the size of the data :) It's an LTV struct (length-type-value), but I'll add a comment